megamek/src/megamek/client/ui/swing/dialog/imageChooser/CamoRenderer.java

```diff
@@ -0,0 +1,109 @@
+/*  
+* MegaMek - Copyright (C) 2020 - The MegaMek Team  
+*  
+* This program is free software; you can redistribute it and/or modify it under  
+* the terms of the GNU General Public License as published by the Free Software  
+* Foundation; either version 2 of the License, or (at your option) any later  
+* version.  
+*  
+* This program is distributed in the hope that it will be useful, but WITHOUT  
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS  
+* FOR A PARTICULAR PURPOSE. See the GNU General Public License for more  
+* details.  
+*/  
+package megamek.client.ui.swing.dialog.imageChooser;
+
+import java.awt.Component;
+
+import javax.swing.BoxLayout;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.ListCellRenderer;
+import javax.swing.UIManager;
+import javax.swing.border.EmptyBorder;
+
+import megamek.client.ui.swing.tileset.CamoManager;
+import megamek.common.util.fileUtils.DirectoryItem;
+
+/** 
+ * A renderer for the list of camos. Shows the camo and the filename
+ * minus the extension below the image. The filename will be shortened
+ * by the JList if it's too long.
+ * Sets the tooltip to show both the directory and the filename of the camo.
+ * 
+ * @author Juliez
+ */
+public class CamoRenderer extends JPanel implements ListCellRenderer<DirectoryItem> {
+
+    private static final long serialVersionUID = -8141491753114750665L;
+    
+    /** Image file extensions, .jpg .jpeg .png .gif */ 
+    private static final String[] EXTENSIONS = {".png", ".jpg", ".jpeg", ".gif"};

```

Comment:
Is this listed anywhere else, and do we support and file types other than these?

Reply:
The ImageFileFactory and ScaledImageFileFactory support only these extensions and have their own Strings for them, but not as an array. The worst that can happen here is that the file extension is shown instead of hidden.